app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/ApplicationController.scala

```diff
@@ -74,6 +73,7 @@ class ApplicationController @Inject() (
         case SpecificGroupIdEnrolmentExists(service) =>
           // Below redirect should be to page that doesn't exists yet
           Redirect(routes.EnrolmentAlreadyExistsController.enrolmentAlreadyExistsForGroup(service))
+        case MissingGroupId() => BadRequest(errorPage())
```

Comment:
Do we have to handle the exception ourselves?  I assumed that would be done for use.

Reply:
yeah, once we start doing `recover` block, the exception if not handled in the recovery, won't be handled. Don't know why

I'm not able to test this manually, so not sure about the behaviour, logic covered by tests. required handling here.

Reply:
You're right, my fault during writing tests before